### PR TITLE
Refactor nav bar on settings screens

### PR DESCRIPTION
**Pull request by devinjameson**
Why: prior to this commit, we were using a custom nav bar component for
navigation on the settings screens.

This commit: updates the nav bar to use React Navigation instead.

## Before
before

## After
after

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -530,4 +530,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: d641f53b267b5fdc2398388c8211cf0739401d29
 
-COCOAPODS: 1.9.2
```
Comment (aledustet) [inline ios/Podfile.lock]: Should we document the why of this downgrade?

File: app/Entry.js [hunk 2]
```diff
@@ -101,15 +101,55 @@ const SelfAssessmentStack = () => (
...
 const MoreTabStack = () => (
-  <Stack.Navigator screenOptions={SCREEN_OPTIONS}>
-    <Stack.Screen name={Screens.Settings} component={SettingsScreen} />
-    <Stack.Screen name={Screens.About} component={AboutScreen} />
-    <Stack.Screen name={Screens.Licenses} component={LicensesScreen} />
-    <Stack.Screen name={Screens.FeatureFlags} component={FeatureFlagsScreen} />
-    <Stack.Screen name={Screens.Import} component={ImportScreen} />
-    <Stack.Screen name={Screens.ImportFromUrl} component={ImportFromUrl} />
-    <Stack.Screen name={Screens.ENDebugMenu} component={ENDebugMenu} />
+  <Stack.Navigator>
```
Comment (aledustet) [inline app/Entry.js]: Solid! Is it possible to add as the default options the `moreTabScreenOptions` and override the two that don't have the same settings with an empty set? Is probably not important but the diff might get smaller. 😁

**Review by aledustet — APPROVED** [review]
👏 💪 
We might want to check the cocoa pods version but other than that is ready to 🚢

File: ios/Podfile.lock (hunk 1 above)
Comment (devinjameson) [reply ios/Podfile.lock]: Can you actually help me understand why this happened?

File: app/Entry.js (hunk 2 above)
Comment (devinjameson) [reply app/Entry.js]: Totally! Good idea.

File: ios/Podfile.lock (hunk 1 above)
Comment (aledustet) [reply ios/Podfile.lock]: Oh it's probably an outdated version cocoapods 🤔 maybe running `gem install cocoapods` will take care of it?

File: app/Entry.js [hunk 3]
```diff
@@ -12,7 +12,7 @@ import {
 } from '@react-navigation/stack';
 import { useSelector } from 'react-redux';
 
-import SettingsScreen from './views/Settings';
```
Comment (johnschoeman) [inline app/Entry.js:14]: 🥇

File: app/views/More/ENLocalDiagnosisKeyScreen.tsx [hunk 4]
```diff
@@ -69,14 +68,8 @@ export const ENLocalDiagnosisKeyScreen = ({
...
 
-  const backToDebugMenu = () => {
-    navigation.goBack();
-  };
-
   return (
-    <NavigationBarWrapper
-      title={'Local Diagnosis Keys'}
-      onBackPress={backToDebugMenu}>
+    <>
```
Comment (johnschoeman) [inline app/views/More/ENLocalDiagnosisKeyScreen.tsx]: can this fragment be removed?

**Review by johnschoeman — APPROVED** [review]
💯 this is a huge improvement.  I would just double check that everything works as intended on the gps build as well. for context we are planning a gps release next week and we need to be diligent about not introducing regressions.

File: app/views/More/ENLocalDiagnosisKeyScreen.tsx (hunk 4 above)
Comment (devinjameson) [reply app/views/More/ENLocalDiagnosisKeyScreen.tsx]: Good catch!

**Comment by johnschoeman** [comment]
more activity

**Comment by aledustet** [comment]
@johnschoeman the rebase was not as smooth as I expected, would you mind having a look?

**Pull request closed without merging**